src/main/app.ts

```diff
@@ -323,6 +403,124 @@ class JupyterApplication implements IApplication, IStatefulService {
             </script>
             </body>
         `;
+        const pageSource = ejs.render(template, {message, checkForUpdatesAutomatically});
+        dialog.loadURL(`data:text/html;charset=utf-8,${pageSource}`);
+    }
+
+    private _showPythonSelectorDialog(reason: 'change' | 'invalid-setting' = 'change') {
+        const dialog = new BrowserWindow({
+            title: 'Set Python Environment',
+            width: 600,
+            height: 250,
+            resizable: false,
+            parent: this._window,
+            modal: true,
+            webPreferences: {
+                nodeIntegration: true,
+                enableRemoteModule: true,
+                contextIsolation: false
+            }
+        });
+        dialog.setMenuBarVisibility(false);
+
+        const bundledPythonPath = this._registry.getBundledPythonPath();
+        const pythonPath = this._applicationState.pythonPath;
+        let useBundledPythonPath = false;
+        if (pythonPath === '' || pythonPath === bundledPythonPath) {
+            useBundledPythonPath = true;
+        }
+        const configuredPath = pythonPath === '' ? bundledPythonPath : pythonPath;
+        const requirements = this._registry.getRequirements();
+        const reqVersions = requirements.map((req) => `${req.name} ${req.versionRange.format()}`);
+        const reqList = reqVersions.join(', ');
+
+        const message = reason === 'change' ?
+            `Select the Python executable in the conda or virtualenv environment you would like to use for JupyterLab Desktop. Python packages in the environment selected need to meet the following requirements: ${reqList}.` :
```

Comment:
Either here or in documentation we should mention that this changes which prebuilt extensions get loaded.

```suggestion
            `Select the Python executable in the conda or virtualenv environment you would like to use for JupyterLab Desktop. Python packages in the environment selected need to meet the following requirements: ${reqList}. Please note that prebuilt extensions installed in selected environment will be loaded.` :
```